frontend/public/components/secrets/create-secret.tsx

```diff
@@ -139,25 +166,30 @@ const generateSecret = () => {
 class WebHookSecretSubform extends React.Component<WebHookSecretSubformProps, WebHookSecretSubformState> {
   constructor(props) {
     super(props);
-    this.state = {WebHookSecretKey: this.props.stringData.WebHookSecretKey || ''};
+    this.state = {
+      stringData: {
+        WebHookSecretKey: this.props.stringData.WebHookSecretKey || ''
+      },
+      authenticationType: SecretType.opaque,
+    };
     this.changeWebHookSecretkey = this.changeWebHookSecretkey.bind(this);
     this.generateWebHookSecret = this.generateWebHookSecret.bind(this);
   }
   changeWebHookSecretkey(event) {
     this.setState({
-      WebHookSecretKey: event.target.value
+      stringData: { WebHookSecretKey: event.target.value }
     }, () => this.props.onChange(this.state));
   }
   generateWebHookSecret() {
     this.setState({
-      WebHookSecretKey: generateSecret()
+      stringData: { WebHookSecretKey: generateSecret() }
     }, () => this.props.onChange(this.state));
   }
   render () {
     return <div className="form-group">
       <label className="control-label" htmlFor="webhook-secret-key">Webhook Secret Key</label>
       <div className="input-group">
-        <input className="form-control" id="webhook-secret-key" type="text" name="webhookSecretKey" onChange={this.changeWebHookSecretkey} value={this.state.WebHookSecretKey} required/>
+        <input className="form-control" id="webhook-secret-key" type="text" name="webhookSecretKey" onChange={this.changeWebHookSecretkey} value={this.state.stringData.WebHookSecretKey} required/>
```

Comment:
Do we need to worry about ensuring `id` attribs are always unique, with something like `_.uniqueId('webhook-secret-key-')`?  We ran into this error a lot in the old console, where a component initially only appeared on the page once... then later would get reused & we would have `id` conflicts.  I'd suggest we just always ensure its unique.